src/lcm/create_params.py

```diff
@@ -102,12 +102,12 @@ def _create_shock_params(model, variable_info, grids):
             inspect.signature(model["functions"][f"next_{var}"]).parameters,
         )
 
-        _check_variables_are_all_discrete_states(
+        _check_variables_are_all_discrete_or_period(
             variables=dependencies,
             variable_info=variable_info,
             msg_suffix=(
-                f"The function next_{var} can only depend on discrete state variables "
-                f"or '_period'."
+                f"The function next_{var} can only depend on discrete variables or"
```

Comment:
Just a clarification because there seems to be a discrepancy to me between this line and the PR title: Does it also include derived variables? 

Say state and choice jointly determine whether a child is born during the period -- but that is neither a (beginning-of-period) state nor a choice. Would that be allowed for?

Reply:
Currently, this is not allowed. Let me check whether this is a straightforward extension.

Reply:
No, it would not be straightforward. We would have to implement a lot more changes. For example, internally, we only keep track of "pure" model variables. We don't have information on derived variables. There are also some other problems: since we are using the index of the variables to access the correct entry in the transition matrix, the transition matrix for derived variables could look very weird. Think `health` in `{0, 1, 2}`, `partner` in `{0, 1, 2}` and `well-being = health * partner`. Then `well-being` in `{0, 1, 2, 4}`.

Reply:
> However, one could simulate this behavior by making the state transition dependent on both variables.

That's how we currently do it in our project: Whether a child is born depends on the derived variable `n_children` which is derived from the state variables `n_children_lagged` and `age_youngest_child` (indicating whether a child is born this period).

However, this leads to unnecessary large and difficult-to-handle transition matrices.

Reply:
Cool. We should discuss in detail at some point whether this will be worth it or not!

Reply:
I am adding it to my LCM notes.